src/website.py

```diff
@@ -63,5 +67,56 @@ def compute_similarity_of_files() -> str:
     return render_template("displaySimilarity.html", text1=texts_start[0], 
                            text2=texts_start[1], text=text)
 
+#navigate to grant call form when button is clicked
+@app.route('/grantcall', methods=['POST', 'GET'])
+def navigateToGrantCallForm() -> str:
+    return render_template("grantCallForm.html")
+
+#calculate similarity and list k most similar scientist upon button click 
+@app.route('/grantcallResult', methods=['POST', 'GET'])
+def calculateGrantCallResult() -> str:
+    #k most similar scientists to display
+    k = 10
+    #get grant call text
+    grantCall = request.form["grantCall"]
+    #calculate embedding for grant call text
+    try:
+        grantCallEmbedding = embedding_from_string(grantCall)
+    except RetryError:
+        return render_template("displayGrantCallResult.html"
+                                   , text1="Please enter a text.")
+
+    #get embeddings from publication list file
+    with h5py.File("../data/pub_embeddings.h5", 'r') as hdf:
+        pub_embedding = hdf['publication_embedding'][:]
+        author_ids = hdf['author_ids'][:]
+    
```

Comment:
This is outdated.  "embeddings" "ids"
![image](https://github.com/mslehre/text-embedding/assets/6883366/faac7511-7136-4554-a4ea-d7a77021fa5f)